pkg/ipcache/cidr.go

```diff
@@ -70,6 +70,13 @@ func (ipc *IPCache) AllocateCIDRs(
 	ipc.Unlock()
 	ipc.metadata.RUnlock()
 
+	// Insert any newly allocated identities in to the policy engine
+	addedIdentities := make(map[identity.NumericIdentity]labels.LabelArray, len(newlyAllocatedIdentities))
+	for _, id := range newlyAllocatedIdentities {
+		addedIdentities[id.ID] = id.LabelArray
+	}
+	ipc.UpdatePolicyMaps(context.TODO(), addedIdentities, nil)
```

Comment:
Incidentally, is this necessary? It seems like the service sync already triggers policy updates:

https://github.com/cilium/cilium/blob/0ff8bdb290a43b81e27d6b1bb5eb8c0f61d0f1d4/pkg/k8s/watchers/watcher.go#L663C19-L663C19

Reply:
I believe so, yes. That TriggerPolicy is the second half of a policy update, but doesn’t add the identities in to the SelectorCache. We still need to do that.

Reply:
Yes, I'm right; the `selectorcache.UpdateIdentities()` call was previously in the allocator event handler, now it's exclusively the domain of the ipcache. We need to trigger that via the `ipc.UpdatePolicyMaps()` call here.